kv/memberlist/tcp_transport.go

```diff
@@ -147,6 +147,9 @@ func NewTCPTransport(config TCPTransportConfig, logger log.Logger, registerer pr
 	port := config.BindPort
 	for _, addr := range config.BindAddrs {
 		ip := net.ParseIP(addr)
+		if ip == nil {
```

Comment:
Another option would be to special case "localhost" here, and resolve that IP address specifically but no others.

Reply:
I think having no special cases here is fine.